app/src/main/java/org/fossasia/pslab/communication/sensors/Sx1276.java

```diff
@@ -0,0 +1,454 @@
+    }
+
+    private void setCodingRate4(int denominator) throws IOException {
+        if (denominator < 5) {
+            denominator = 5;
+        } else if (denominator > 8) {
+            denominator = 8;
+        }
+        setupBytesArray((SPIRead(REG_MODEM_CONFIG_1, 1).get(0) & 0xF1) | ((denominator - 4) << 4));
+        SPIWrite(REG_MODEM_CONFIG_1, bytes);
+    }
+
+    public void setPreambleLength(int length) throws IOException {
+        setupBytesArray((length >> 8) & 0xFF);
+        SPIWrite(REG_PREAMBLE_MSB, bytes);
+        setupBytesArray(length & 0xFF);
+        SPIWrite(REG_PREAMBLE_LSB, bytes);
+    }
+
+    public void setSyncWord(ArrayList<Byte> Word) throws IOException {
+        SPIWrite(REG_SYNC_WORD, Word);
+    }
+
+    public void crc() throws IOException {
+        setupBytesArray(SPIRead(REG_MODEM_CONFIG_2, 1).get(0) | 0x04);
+        SPIWrite(REG_MODEM_CONFIG_2, bytes);
+    }
+
+    public void noCrc() throws IOException {
+        setupBytesArray(SPIRead(REG_MODEM_CONFIG_2, 1).get(0) & 0xFB);
+        SPIWrite(REG_MODEM_CONFIG_2, bytes);
+    }
+
+    public byte random() throws IOException {
+        return (SPIRead(REG_RSSI_WIDEBAND, 1).get(0));
+    }
+
+    private void explicitHeaderMode() throws IOException {
+        implicitHeaderMode = 0;
+        setupBytesArray(SPIRead(REG_MODEM_CONFIG_1, 1).get(0) & 0xFE);
+        SPIWrite(REG_MODEM_CONFIG_1, bytes);
+    }
+
+    private void implicitHeaderMode() throws IOException {
+        implicitHeaderMode = 1;
+        setupBytesArray(SPIRead(REG_MODEM_CONFIG_1, 1).get(0) | 0x01);
+        SPIWrite(REG_MODEM_CONFIG_1, bytes);
+    }
+
+    public void handleDio0Rise() throws IOException {
+        byte irqFlags = SPIRead(REG_IRQ_FLAGS, 1).get(0);
+        setupBytesArray(irqFlags);
+        SPIWrite(REG_IRQ_FLAGS, bytes);
+
+        if ((irqFlags & IRQ_PAYLOAD_CRC_ERROR_MASK) == 0) {
+            packetIndex = 0;
+        }
+        if (implicitHeaderMode == 0) {
+            packetLength = SPIRead(REG_PAYLOAD_LENGTH, 1).get(0);
+        } else {
+            packetLength = SPIRead(REG_RX_NB_BYTES, 1).get(0);
+        }
+
+        SPIWrite(REG_FIFO_ADDR_PTR, SPIRead(REG_FIFO_RX_CURRENT_ADDR, 1));
+        if (onReceive == 1) {
+            Log.d(name, "Packet Length " + packetLength);
+        }
+
+        setupBytesArray(0);
+        SPIWrite(REG_FIFO_ADDR_PTR, bytes);
+    }
+
+    private ArrayList<Byte> SPIWrite(int adr, ArrayList<Byte> byteArray) throws IOException {
+        // CS1 => 9
+        ArrayList<Byte> data = new ArrayList<>();
+        data.add((byte) (0x80 | adr));
+        data.addAll(byteArray);
+        ArrayList<Byte> XFER = spi.xfer(9, data);
+        XFER.remove(0);
+        return XFER;
+    }
+
+    private ArrayList<Byte> getRaw() throws IOException {
+        return SPIRead(0x02, 1);
+    }
+
+    private void setupBytesArray(int data) {
+        bytes.clear();
+        bytes.add((byte) data);
```

Comment:
This method was added by me to simplify creating a byte array. Since all the values fetched into the parameters are in the range of a byte memory size, it wouldn't cause any errors! If this parameter is set to `byte` then I have to cast integers like `0` when the method is called. So I thought of keeping it as `int` and do the casting inside the method! :)